### DISCUSSION TITLE
How to choose strategies for attribute filtering

### DISCUSSION THREAD

**Discussion by yushangdi** [milvus-io/milvus#31551] [Q&A and General discussion] [answered]
https://www.cs.purdue.edu/homes/csjgwang/pubs/SIGMOD21_Milvus.pdf

The Milvus paper describes 5 strategies in Figure 4. I have a few questions regarding the strategies.
1) If I run my filtered search like below, which strategy is used?
```
# e.g. HNSW, or an IVF index. not FLAT.
points.create_index("data", index_params)
points.load()
result = points.search(
                    queries,
                    "data",
                    search_params,
                    limit=TOP_K,
                    expr=expr,
                )
```

2) Can I configure the filtered search to use one of the strategies? And if I can, how should I do it for each of the strategies? (especially strategy D and E). 

I checked this [post](https://github.com/milvus-io/milvus/discussions/26841), but still don't quite know the answer. Thanks!

**Comment by xiaofan-luan** [comment, marked as answer]
Currently, opensource milvus use prefiltering or partition prefiltering.

We actually explored a lot of different strategy on improving the performance. One of the interesting policy we will be work on is to add additional graph edge on the filtering condition
